src/pages/api/views/download.ts

```diff
@@ -0,0 +1,177 @@
+    const notes = cell as ZetkinNote[];
+    return notes.length ? notes[0].text : '';
+  }) as ValueFunc,
+  [COLUMN_TYPE.PERSON_QUERY]: directFormatter,
+  [COLUMN_TYPE.PERSON_TAG]: directFormatter,
+  [COLUMN_TYPE.SURVEY_RESPONSE]: ((cell) => {
+    const responses = cell as SurveyResponse[];
+    return responses.length ? responses[0].text : '';
+  }) as ValueFunc,
+  [COLUMN_TYPE.SURVEY_SUBMITTED]: ((cell) => {
+    const submissions = cell as SurveySubmission[];
+    return submissions.length ? submissions[0].submitted : '';
+  }) as ValueFunc,
+};
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> {
+  const { format = 'csv', orgId, viewId } = req.query;
+
+  if (
+    !orgId ||
+    Array.isArray(orgId) ||
+    Array.isArray(viewId) ||
+    Array.isArray(format)
+  ) {
+    return res.status(400).end();
+  }
+
+  if (format != 'csv' && format != 'xlsx') {
+    return res.status(400).end();
+  }
+  const formatType = FORMAT_TYPES[format];
+
+  const apiClient = new BackendApiClient(req.headers);
+
+  const view = await apiClient.get<ZetkinView>(
+    `/api/orgs/${orgId}/people/views/${viewId}`
+  );
+  const columns = await apiClient.get<ZetkinViewColumn[]>(
+    `/api/orgs/${orgId}/people/views/${viewId}/columns`
+  );
+  const rows = await apiClient.get<ZetkinViewRow[]>(
+    `/api/orgs/${orgId}/people/views/${viewId}/rows`
+  );
+
+  const headerRow: string[] = ['ID'].concat(columns.map((col) => col.title));
+
+  const dataRows: CellValue[][] = rows.map((row) =>
+    [row.id as CellValue].concat(
+      row.content.map((cell, index) => {
+        const col = columns[index];
+        return valueFormatters[col.type](cell, col);
+      })
+    )
+  );
+
+  const filePrefix = slugify(view.title).toLowerCase();
+  const fileName = `${filePrefix}-${new Date().toISOString()}.${format}`;
+
+  let fileData: string | Buffer;
+
+  const wb = XLSX.utils.book_new();
+  const sheet = XLSX.utils.aoa_to_sheet([headerRow, ...dataRows], {
+    dateNF: 'YYYY-MM-DD HH:mm:ss',
+  });
+  XLSX.utils.book_append_sheet(wb, sheet);
+
+  if (format == 'csv') {
+    // Make header uppercase
+    headerRow.forEach((_, idx) => {
+      const addr = XLSX.utils.encode_cell({ c: idx, r: 0 });
+      sheet[addr].v = sheet[addr].v.toString().toUpperCase();
+    });
+
+    fileData = XLSX.write(wb, { bookType: 'csv', type: 'buffer' });
+  } else if (format == 'xlsx') {
+    sheet['!cols'] = headerRow.map((col, idx) => {
+      const allLengths = [
+        col.length,
+        ...dataRows.map((row) =>
+          row[idx] instanceof Date ? 16 : row[idx]?.toString().length ?? 0
+        ),
+      ];
+
+      return {
+        // Set the width to 3 + the max number of characters on any row,
+        // but cap to 50 characters to not get super wide columns
+        width: 3 + Math.min(50, Math.max.apply(null, allLengths)),
+      };
+    });
+
+    // Style header as bold
+    headerRow.forEach((_, idx) => {
```

Comment:
It's the little things like width and bold headers that makes it so nice to use!